src/trivialserialize.cpp

```diff
@@ -119,7 +119,9 @@ void parser::pop_byte_skip(char c) { // read expected character (e.g. a delimite
 std::string parser::pop_bytes_n(size_t size) {
 	if (!size) return string("");
 	if (! (m_data_now < m_data_end) ) throw format_error_read(); // we run outside of string
-	if (! (   static_cast<unsigned long long int>(m_data_end - m_data_now) >= size) ) throw format_error_read(); // the read will not fit // TODO(r) is this cast good?
+	// casting below is ok, because std::ptrdiff_t is 64 bits signed value, and size_t is 64 bits unsigned value.
```

Comment:
Ok, BUT TODO @damianbb comment needs clarification to "e.g. 64 bits".
Also I would add static_assert that size of ptrdiff_t >= sizeof size_t just in case.